src/components/scene/background.js

```diff
@@ -15,6 +16,30 @@ module.exports.Component = register('background', {
       return;
     }
     object3D.background = new THREE.Color(data.color);
+
+    if (data.generateEnvironment) {
+      const scene = this.el.sceneEl.object3D;
+      const cubeRenderTarget = new THREE.WebGLCubeRenderTarget(128, { format: THREE.RGBFormat, generateMipmaps: true, minFilter: THREE.LinearMipmapLinearFilter });
+      const cubeCamera = new THREE.CubeCamera(1, 100000, cubeRenderTarget);
+      this.cubeCamera = cubeCamera;
+      this.needsEnvironmentUpdate = true;
+      scene.environment = cubeRenderTarget.texture;
+      this.el.sceneEl.addEventListener('loaded', function () {
```

Comment:
I believe this event handler is not needed.

Reply:
I think you are right,

On a related note, do you know if there is an event handler for things like an object or image has finished loading asynchronously, which would allow this to still work nicely even if someone is loading their skybox etc without using `<a-assets>`?

Reply:
mmh not sure if there's a way to universally know it all assets have loaded if `<a-assets>` is not used. Components can do all sort of arbitrary stuff.

Reply:
I guess if someone wanted this behaviour they could do `background.needsEnvironmentUpdate=true` on this component from their custom components.

It might be nice if components like `material` would fire a bubbling event once they had loaded an image asynchronously so components like this can update the texture in a few common situations. Although this feature may be the carrot needed to get more developers to use `<a-assets>`